account_invoice_import/wizard/account_invoice_import.py

```diff
@@ -0,0 +1,1456 @@
+        # to avoid "Object of type bytes is not JSON serializable"
+        assert invoice_file_b64
+        if isinstance(invoice_file_b64, str):
+            invoice_file_b64 = invoice_file_b64.encode("utf8")
+        assert isinstance(invoice_file_b64, bytes)
+        assert isinstance(invoice_filename, str)
+        aiico = self.env["account.invoice.import.config"]
+        if company_id is None:
+            company_id = self.env.company.id
+        logger.info(
+            "Starting to import invoice file %s in company ID %d",
+            invoice_filename,
+            company_id,
+        )
+        parsed_inv = self.parse_invoice(invoice_file_b64, invoice_filename)
+        partner = self._match_partner(parsed_inv["partner"], parsed_inv["chatter_msg"])
+        partner = partner.commercial_partner_id
+        existing_inv = self.invoice_already_exists(partner, parsed_inv)
+        if existing_inv:
+            logger.warning(
+                "This supplier invoice already exists "
+                "in Odoo (ID %d number %s supplier number %s)",
+                existing_inv.id,
+                existing_inv.name,
+                parsed_inv.get("invoice_number"),
+            )
+            return False
+        import_configs = aiico.search(
+            [("partner_id", "=", partner.id), ("company_id", "=", company_id)]
+        )
+        if not import_configs:
+            logger.warning(
+                "Missing invoice import configuration "
+                "for partner '%s' in company ID %d.",
+                partner.display_name,
+                company_id,
+            )
+            return False
+        elif len(import_configs) == 1:
+            import_config = import_configs.convert_to_import_config()
+        else:
+            logger.info(
+                "There are %d invoice import configs for partner %s in company ID %d. "
+                "Using the first one '%s''",
+                len(import_configs),
+                partner.display_name,
+                company_id,
+                import_configs[0].name,
+            )
+            import_config = import_configs[0].convert_to_import_config()
+        invoice = self.create_invoice(parsed_inv, import_config, origin)
+        return invoice.id
+
+    @api.model
+    def _prepare_global_adjustment_line(self, diff_amount, invoice, import_config):
+        amlo = self.env["account.move.line"]
+        prec = invoice.currency_id.rounding
+        sign = diff_amount > 0 and 1 or -1
+        il_vals = {
+            "name": _("Adjustment"),
+            "quantity": sign,
+            "price_unit": diff_amount * sign,
+        }
+        # no taxes nor product on such a global adjustment line
+        if import_config["invoice_line_method"] == "nline_no_product":
+            il_vals["account_id"] = import_config["account"].id
+        elif import_config["invoice_line_method"] == "nline_static_product":
+            account = amlo.get_invoice_line_account(
```

Comment:
Can you include a fix for:
https://github.com/OCA/edi/issues/572

It is a leftover of V14 migration.